lib/src/mock_document_reference.dart

```diff
@@ -119,20 +122,19 @@ class MockDocumentReference extends Mock implements DocumentReference {
     return Stream.value(
         MockDocumentSnapshot(this, _documentId, root, _exists()));
   }
+}
 
-  static Map<String, dynamic> _deepCopy(Map<String, dynamic> fromMap) {
-    final toMap = <String, dynamic>{};
-
-    fromMap.forEach((key, value) {
-      if (value is Map<String, dynamic>) {
-        toMap[key] = _deepCopy(value);
-      } else if (value is List) {
-        toMap[key] = List.from(value);
-      } else {
-        toMap[key] = value;
-      }
+/// Copies data replicating its inner Maps and Lists.
+dynamic deepCopy(dynamic fromData) {
+  if (fromData is Map<String, dynamic>) {
+    final toMap = Map<String, dynamic>.from(fromData);
+    toMap.forEach((key, value) {
+      toMap[key] = deepCopy(value);
     });
-
     return toMap;
+  } else if (fromData is List) {
+    return fromData.map((v) => deepCopy(v)).toList();
```

Comment:
Can you write it as `fromData.map(deepCopy).toList();`?

Reply:
Nice. Updated.